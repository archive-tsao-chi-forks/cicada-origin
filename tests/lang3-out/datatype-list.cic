@datatype List : (Type) -> Type {
  null: (T: Type) -> List(T)
  cons: (T: Type) -> (head: T) -> (tail: List(T)) -> List(T)
}

// @show List
