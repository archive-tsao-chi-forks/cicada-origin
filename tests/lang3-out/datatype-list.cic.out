@datatype List : (_: Type) -> Type {
  null : (T: Type) -> List(T)
  cons : (T: Type) -> (head: T) -> (tail: List(T)) -> List(T)
}
List.null
List.cons
null(String)
cons(String)("abc")(null(String))

